mapit_global/management/commands/mapit_global_update_countries.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import print_function, unicode_literals
+
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from django.utils.functional import cached_property
+from lxml import etree
+import requests
+
+from mapit.models import Area, Code, CodeType, Country, Generation
+
+
+class Command(BaseCommand):
+
+    OSM_COUNTRY_CODE_KEY = 'ISO3166-1'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--commit', action='store_true')
+        parser.add_argument('--generation', type=int)
+
+    def generation_kwargs(self):
+        return {
+            'generation_low__lte': self.generation,
+            'generation_high__gte': self.generation,
+        }
+
+    def set_country_codes_for_countries(self):
+        country_areas_qs = Area.objects.filter(
+            type__code='O02', **self.generation_kwargs())
+        for area in country_areas_qs:
+            area.codes.filter(type=self.iso_code_type).delete()
+            code = area.codes.get(type__code__in=('osm_rel', 'osm_way'))
+            osm_type = {
+                'osm_rel': 'relation',
+                'osm_way': 'way',
+            }[code.type.code]
+            api_url = 'http://api.openstreetmap.org/api/0.6/{0}/{1}'.format(osm_type, code.code)
+            print('api_url:', api_url)
+            r = requests.get(api_url)
+            if r.status_code == 410:
```

Comment:
What does getting a 410 mean for us here, why is it ignored?

Reply:
We get a 410 Gone for: http://api.openstreetmap.org/api/0.6/relation/1111111 which has been deleted (it was a duplicate relation for the boundary of Germany). I'll add a comment that some boundaries might have been deleted since the import, which is why that comes up.